### PR TITLE
[NFT-938] feat: target updates in sync with blocks

### PR DESCRIPTION
**Pull request by cnasc**

### CODE REVIEW

File: components/ApplicationProviders/ApplicationProviders.tsx [hunk 1]
```diff
@@ -95,13 +96,13 @@ export const ApplicationProviders: FunctionComponent = ({ children }) => {
           <CenterProvider
             network={centerNetwork as any}
             apiKey={process.env.NEXT_PUBLIC_CENTER_KEY ?? ''}>
-            <TargetProvider>
-              <PaprBalanceProvider>
-                <TimestampProvider>
+            <TimestampProvider>
```
Comment (cnasc) [inline components/ApplicationProviders/ApplicationProviders.tsx:99]: now that target relies on timestamp, we need `TimestampProvider` to be outside `TargetProvider`

File: components/ContractStatus/ContractStatus.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ export function ContractStatus() {
     error: pricesDataError,
   } = useControllerPricesData();
   const contractAPR = useMemo(() => {
-    if (!pricesData) {
+    if (!pricesData || pricesData.targetValues.length < 2) {
```
Comment (cnasc) [inline components/ContractStatus/ContractStatus.tsx:18]: There are now cases where we briefly have an empty array of targets, handled those in this PR as they came up

**Review by wilsoncusack — APPROVED** [review]

**Comment by wilsoncusack** [comment]
does this mean we no longer cache and will make RPC calls more often?

**Comment by cnasc** [comment]
> does this mean we no longer cache and will make RPC calls more often?

We'll call `newTarget` every new block. We can set some threshold and only update if the new timestamp is some amount greater than the old one (e.g. hold off for 2 minutes)

**Pull request merged**